FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc

```diff
@@ -194,59 +153,24 @@ TrajectorySeedProducer::passSimTrackQualityCuts(const SimTrack& theSimTrack, con
             0.,0.,4.
     );
     theParticle.setCharge(theSimTrack.charge());
-
-
-    //this are just some cuts on the SimTrack for speed up
-    std::vector<const math::XYZPoint*> origins;
-    if (testBeamspotCompatibility)
-    {
-        origins.push_back(&beamSpot->position());
-    }
-    if (testPrimaryVertexCompatibility)
-    {
-        for (unsigned int iv = 0; iv < primaryVertices->size(); ++iv)
-        {
-            origins.push_back(&(*primaryVertices)[iv].position());
-        }
-    }
-
-    //only one possible origin is required to succeed
-    for (unsigned int i = 0; i < origins.size(); ++i)
-    {
-        if ((simTrack_maxD0>0.0) && ( theParticle.xyImpactParameter(origins[i]->X(),origins[i]->Y()) > simTrack_maxD0 ))
-        {
-            continue;
-        }
-        if ((simTrack_maxZ0>0.0) && ( fabs( theParticle.zImpactParameter(origins[i]->X(),origins[i]->Y()) - origins[i]->Z()) > simTrack_maxZ0))
-        {
-            continue;
-        }
-        return true;
-    }
+    if ((simTrack_maxD0>0.0) && ( theParticle.xyImpactParameter() > simTrack_maxD0 )&&((simTrack_maxZ0>0.0) && ( fabs( theParticle.zImpactParameter()) > simTrack_maxZ0)))
+	  {
+	    return true;
+	   }
+    
     return false;
 }
 
 bool
 TrajectorySeedProducer::pass2HitsCuts(const TrajectorySeedHitCandidate& hit1, const TrajectorySeedHitCandidate& hit2) const
 {
-
-    const GlobalPoint& globalHitPos1 = hit1.globalPosition();
-    const GlobalPoint& globalHitPos2 = hit2.globalPosition();
-    bool forward = hit1.isForward(); // true if hit is in endcap, false = barrel
-    double error = std::sqrt(hit1.largerError()+hit2.largerError());
-    if (testBeamspotCompatibility)
-      {
-	return compatibleWithBeamSpot(globalHitPos1,globalHitPos2,error,forward);
-      }
-    else if(testPrimaryVertexCompatibility)
-      {
-	return compatibleWithPrimaryVertex(globalHitPos1,globalHitPos2,error,forward);
-      }
-    else
-      {
-	return true;
-      }
- 
+  if(theRegionProducer){
+  return testWithRegions(hit1,hit2);	                 
```

Comment:
please move the content of testWithRegions inside pass2HitsCuts and remove the original testWithRegions